src/main/java/com/google/growpod/servlets/GardenServlet.java

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.growpod.servlets;
+
+import com.google.growpod.data.Garden;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that handles garden entities on the server.
+ *
+ * <p>API DOCUMENTATION: /garden/{id} {id} -- A garden UUID GET: Retrieves the garden data structure
+ * for {id} No parameters Returns data in JSON format along with (200 OK), otherwise (404 NOT FOUND)
+ * /garden/{id}/plant-list {id} -- A garden UUID GET: Retrieves all plants currently on {id} garden
+ * No parameters Returns data in JSON format along with (200 OK), otherwise (404 NOT FOUND)
+ * /garden/{id}/user-list {id} -- A garden UUID GET: Retrieves all users currently on {id} garden No
```

Comment:
How do you associate users with gardens?

Reply:
Right now, everything's hardcoded, but in the real database, I will most likely have a table with one column being the User ID and the other being the Garden ID.

And then there are the two API calls, `GET: /garden/{id}/user-list` and `GET: /user/{id}/garden-list`, to get the users that are members of garden {id} and get the gardens that the user {id} is a member of respectively.